packages/pds/src/sql-repo-storage.ts

```diff
@@ -0,0 +1,200 @@
+import { RepoStorage } from '@atproto/repo'
+import { CID } from 'multiformats/cid'
+import Database from './db'
+import { IpldBlock } from './db/tables/ipld-block'
+import { IpldBlockCreator } from './db/tables/ipld-block-creator'
+import { RepoCommitBlock } from './db/tables/repo-commit-block'
+
+export class SqlRepoStorage extends RepoStorage {
+  constructor(
+    public db: Database,
+    public did: string,
+    public timestamp?: string,
+  ) {
+    super()
+  }
+
+  async getHead(forUpdate?: boolean): Promise<CID | null> {
+    let builder = this.db.db
+      .selectFrom('repo_root')
+      .selectAll()
+      .where('did', '=', this.did)
+    if (forUpdate) {
+      this.db.assertTransaction()
+      if (this.db.dialect !== 'sqlite') {
+        // SELECT FOR UPDATE is not supported by sqlite, but sqlite txs are SERIALIZABLE so we don't actually need it
+        builder = builder.forUpdate()
+      }
+    }
+    const found = await builder.executeTakeFirst()
+    return found ? CID.parse(found.root) : null
+  }
+
+  async getSavedBytes(cid: CID): Promise<Uint8Array | null> {
+    const found = await this.db.db
+      .selectFrom('ipld_block')
+      .where('cid', '=', cid.toString())
```

Comment:
Should `getSavedBytes()` and `hasSavedBytes()` be restricted by `this.did`?

Reply:
yeah that's a good idea